packages/lib/server-only/template/create-document-from-template.ts

```diff
@@ -256,10 +256,20 @@ export const createDocumentFromTemplate = async ({
       },
     });
 
+    const recipientMapping = new Map<number, number>();
+
+    template.recipients.forEach((templateRecipient, index) => {
```

Comment:
Connects template recipient IDs to their corresponding document recipient IDs, so fields can be correctly assigned to the right recipients in the new document.